src/file_tree.ml

```diff
@@ -135,13 +136,19 @@ module Dir = struct
     String.Map.foldi (sub_dirs t) ~init:Path.Source.Set.empty
       ~f:(fun s _ acc -> Path.Source.Set.add acc (Path.Source.relative t.path s))
 
-  let rec fold t ~traverse_ignored_dirs ~init:acc ~f =
-    if not traverse_ignored_dirs && t.ignored then
+  let skip ~traverse_ignored_dirs ~traverse_vendored_dirs t =
+    match t.status with
+    | Data_only -> not traverse_ignored_dirs
```

Comment:
It's not related to this PR, but we should rename this variable to `traverse_data_only_dirs`. This would be more in line with the status name.

Reply:
Fixed it!

Reply:
Thanks!